### PR TITLE
[6.x] Fix migration overwrite $connection property

### LINKED ISSUE #31060: Migration overwrite $connection property don't work
- Laravel Version: 6.10.0
- PHP Version: 7.2.18
- Database Driver & Version: Mysql - 5.7

### Description:
I overwrite $connection property in migration, but still migrate on the default database connection

as its described on the "abstract class Migration"

```php
abstract class Migration
{
    /**
     * The name of the database connection to use.
     *
     * @var string|null
     */
    protected $connection;
```

`The migration file i used`

```php
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

class CreateProjectsTable extends Migration
{
    protected $connection = 'pgsql';
    
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::create('projects', function (Blueprint $table) {
            $table->bigIncrements('id');
            $table->timestamps();
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::dropIfExists('projects');
    }
}
```

Note: the default connection i use on .env is mysql

### PR DESCRIPTION
**Pull request by 4m1n0s**
Closes #31060.

### CODE REVIEW

File: src/Illuminate/Database/Migrations/Migrator.php [hunk 1]
```diff
@@ -524,6 +526,24 @@ public function getConnection()
...
     }
+    
+    /**
+     * Execute the given callback using the given connection as the default connection.
+     *
+     * @param  string  $name
+     * @param  callable  $callback
+     * @return mixed
+     */
+    public function usingConnection($name, callable $callback)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Migrations/Migrator.php:537]: Which PR was this method originally introduced in?

Comment (iamine) [reply src/Illuminate/Database/Migrations/Migrator.php:537]: Here: https://github.com/laravel/framework/commit/f4bd05fc1015a1a49dd6850d6817a10d129143c8

**Comment by GrahamCampbell** [comment]
I don't think 6.x is the right place for this fix anymore tbh, since we don't want to partially apply some of that other stuff that only landed in 7.x.

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!

**Pull request closed without merging**